### ISSUE TITLE
Weird button states since 7.4.7

### ISSUE THREAD

**Issue by ZusYar** [renpy/renpy#3181]
Buttons' behavior changed, from my perspective, in a bad way.
Example: i have an image with a set of tiles that represent a button's states: idle, hover, active, etc

I could simply program the buttons with a transform like
```
transform tile (x, y, w, h):
    'tiles.webp'
    crop (x * w, y * h, w, h)
```
```
    imagebutton:
        idle tile (0, 0, 100, 20)
        hover tile (200, 80, 100, 20)
        insensitive tile (100, 180, 100, 20)
        action NullAction()
```
It worked perfectly and I could change tiles programmatically on key pressed and all
Well, not now. Buttons won't get idle state ever again after you unhover them
```
button:
    add tiles at some_cropping_transform()
```
doesn't work either, because whenever the button is insensitive it doesn't get transformed at all.
I'm not sure how i'm supposed to create such buttons anymore.
This is very frustrating, and it's practically impossible (tedious) to migrate a ready project to newer renpy.
(tested every major version from 7.3.5 up to 7.4.10. Any version before 7.4.7 handles idle state very well.)

So the question is: is this behavior intentional?
Will this ever work again?

**Comment by Andykl** [comment]
Confirm it happens, even in 8.0
Example code
```py
label main_menu:
    $ _confirm_quit = False
    return

image atlas_like = VBox(
    Fixed(Solid("#0ff"), Text("Idle", align=(0.5, 0.5)), xysize=(200, 100)),
    Fixed(Solid("#ff0"), Text("Hover", align=(0.5, 0.5)), xysize=(200, 100)),
)

transform tile (x, y, w, h):
    'atlas_like'
    crop (x, y, w, h)

screen test():
    imagebutton:
        align (0.5, 0.5)
        idle tile (0, 0, 200, 100)
        hover tile (0, 100, 200, 100)
        action NullAction()

label start:
    hide black
    show screen test
    pause
    return
```

**Comment by Andykl** [comment]
If `tile` call is wrapped into `ImageReference` problem is gone.

**Comment by Andykl** [comment]
@ZusYar This is a feature/regression introduced in 65d719ecacd4143416dae6c8080b7bc4c39af38f to fix #2864.
Good thing is that if you wrap Transform call into ImageReference via `image` statement or by just
```
init python:
    def tile(x, y, w, h):
        return ImageReference(tile_(x, y, w, h))

transform tile_ (x, y, w, h):
    'tiles.webp'
    crop (x * w, y * h, w, h)
```
your problem is fixed.

I do not know if that change should be considered a regression. As said in 9c0e7148dec16eef55705974f054a58b3910fa39 it is a compromise, so probably will be fixed at some point.
